### PR TITLE
On RecordTable, if I have no records, the Record Table Layout is not broken

### LINKED ISSUE #2466: On RecordTable, if I have no records, the Record Table Layout is not broken
We are experiencing issues with our RecordTable Layout. The root cause of the problem is that we want two behaviors that are incompatible with display: table-cell behaviors:

- If the columns are larger than the screen, we want the + button to add new viewFields to be visible at the end: 
image ==> this means we want a fixed width

- if the columns are smaller than the screen, we want the + button to take the whole width:
image
 ==> this means we want a 100% width

In display: block css we would do a width: 100% + min-width: 32px.
However this is incompatible with table-cell behavior that does not support it.

We have hacked it to work but it does not work when the table is loading:
image

The proposal is to keep a table-cell display as it is BUT to compute the + column size programmatically in javascript. This should be doable in RecordTableHeader using Refs and getting width of the table and columns.

## Alternative solutions:
We could also decide to not use the table-cell display in this case but this would break other nice behaviors (alignment, nice border collapsing...) and make it complex.

### PR DESCRIPTION
**Pull request by gitstart-twenty**
Fixes #2466

### CODE REVIEW

**Comment by lucasbordeau** [comment]
Nice,

We need to have a bottom line on the last "+" column : 

image

**Comment by gitstart-twenty** [comment]
Hey @lucasbordeau 
This has been updated

**Comment by charlesBochet** [comment]
@gitstart-twenty I think there is a last case, when I have no additional field to show and I have no data:
image

**Review by gitstart-twenty — COMMENTED** [review]
Hey @charlesBochet,
The default behavior was to have the button only show when there were hidden columns. I've disabled this.

**Comment by charlesBochet** [comment]
@gitstart-twenty Actually, we don't want to show the button when there is no other fields to show but when the button is not shown, the column width should also be respected and we should show an extra empty column at the end.
So maybe, always show the additionnal empty column (if table width < scroll view width) but only show the plus if there are additional columns to show

**Comment by gitstart-twenty** [comment]
Hey @charlesBochet Update here:

https://github.com/twentyhq/twenty/assets/140154534/510f934c-5b29-4fde-a683-85445658f8cf

File: packages/twenty-front/src/modules/object-record/record-table/components/RecordTableHeader.tsx [hunk 1]
```diff
@@ -55,6 +63,12 @@ export const RecordTableHeader = ({
     useRecordTableScopedStates();
 
   const hiddenTableColumns = useRecoilValue(hiddenTableColumnsSelector);
+
+  const scrollWrapper = useContext(ScrollWrapperContext);
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-table/components/RecordTableHeader.tsx:67]: Let's use useScrollWrapperScopedRef here

**Review by charlesBochet — APPROVED** [review]
This is working as intended! Fixing a very old bug, nice!

**Pull request merged**